src/lib/services/licences.js

```diff
@@ -108,13 +110,29 @@ const updateIncludeInSupplementaryBillingStatusForSentBatch = async batchId => {
   return updateIncludeInSupplementaryBillingStatusForUnsentBatch(batchId);
 };
 
+/**
+ * Fetches the invoice accounts associated with a licence using the licence ref and date as input.
+ *  @param {String} documentRef
+ *  @param {String} date
+ */
+const getLicenceAccountsByRefAndDate = async (documentRef, date) => {
+  //  First, fetch the company ID from the CRM
+  const { companyId } = await crmDocsConnector.getDocumentByRefAndDate(documentRef, date);
+
+  //  Secondly, take the company ID and fetch the invoice accounts for that company from the CRM
+  const invoiceAccounts = await crmCompaniesConnector.getInvoiceAccountsByCompanyId(companyId);
+
+  //  Return the invoice accounts
+  return invoiceAccounts;
```

Comment:
The invoice accounts should be mapped to the water service models before returning them to the UI and if there are no invoice accounts then return an empty array. The mappers and models already exist in ./src/lib i.e. 

return invoiceAccounts ? InvoiceAccountsMapper.crmToModel(invoiceAccounts) : [];

Reply:
Will do, but I think I'll need that line to be:

`return invoiceAccounts ? invoiceAccounts.map(invoiceAccount => invoiceAccountsMapper.crmToModel(invoiceAccount)) : [];`

This is in order to get the invoice accounts to map to the model one at a time (as it's returning an array of them)

I hope that's OK but if it's not let me know and I'll undo.